File: VonexDealer.API/Models/Order/IPOrder.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Threading.Tasks;
using VonexDealer.API.Models.UB;

namespace VonexDealer.API.Models.Order
{

    public class IPOrder
    {
        public Int64 IPOrderID { get; set; }
        public Int64 OrderID { get; set; }
        public Int64? RatePlanID { get; set; }
        public string CurrentModem { get; set; }
        public string CurrentSwitch { get; set; }
        public string UploadSpeed { get; set; }
        public string DownloadSpeed { get; set; }
        public bool IsCreatedByAMPT { get; set; }
        public string AMPTDomainName { get; set; }
        public CallHandling callHandling { get; set; }
        public bool DealerSupplied { get; set; }
        public bool MinMonthly { get; set; }
    }

    /// <summary>
    /// Call Handling options for form submission.
    /// </summary>
    public enum CallHandling
    {
        [Description("Direct To Extension")]
        DirectToExtension,
        [Description("Hunt Group")]
        HuntGroup ,
        [Description("AutoAttendant (IVR)")]
        AutoAttendant 

    }


}

File: VonexDealer.Vue/ClientApp/models/ipOrder.js
export class ipOrder {
    constructor({
        iPOrderID = 0,
        orderID = 0,
        ratePlanID = 0,
        currentModem = null,
        currentSwitch = null,
        uploadSpeed = null,
        downloadSpeed = null,
        callHandline = null,
        isCreatedByAMPT = false,
        amptDomainName = null,
        dealerSupplied = false,
        minMonthly = false

    } = {}) {
        this.iPOrderID = iPOrderID;
        this.orderID = orderID;
        this.ratePlanID = ratePlanID;
        this.currentModem = currentModem;
        this.currentSwitch = currentSwitch;
        this.uploadSpeed = uploadSpeed;
        this.downloadSpeed = downloadSpeed;
        this.callHandline = callHandline;
        this.isCreatedByAMPT = isCreatedByAMPT;
        this.amptDomainName = amptDomainName;
        this.dealerSupplied = dealerSupplied;
        this.minMonthly = minMonthly;
    }
}

export function CreateIPOrder(data) {
    return Object.freeze(new ipOrder(data))
}
